ddtrace/context.py

```diff
@@ -50,13 +56,28 @@ def get_current_span(self):
         with self._lock:
             return self._current_span
 
+    def _set_current_span(self, span):
+        """
+        Set current span internally.
+
+        Non-safe if not used with a lock. For internal Context usage only.
+        """
+        self._current_span = span
+        if span:
+            self._parent_trace_id = span.trace_id
+            self._parent_span_id = span.span_id
+            self._sampled = span.sampled
+            self._sampling_priority = span.get_sampling_priority()
```

Comment:
Will use the pythonic approach of having a property, AFAIR the getter/setter were here initially because first implementations were actually setting a meta so using this did save a lot of copy/paste. But now that we have this sampling_priority as a class member, they sound obviously over engineered. Removing them.

Reply:
https://github.com/DataDog/dd-trace-py/pull/359/commits/e1257b411df64ab032a30b00315d55a373d1f4a3